tests/slim_recipe_specs.py

```diff
@@ -0,0 +1,51 @@
+"""specify which tests to run on each of the recipes in ./examples"""
+
+import os
+
+# possible attributes to simulation scripts are
+#  WF, nonWF
+#  nucleotides
+#  everyone: records everyone ever
+#  pedigree: writes out accompanying info file containing the pedigree
+#  remembered_early: remembering and saving the ts happens during early
+#  multipop: has more than one population
+# All files are of the form `tests/examples/{key}.slim`
+basic_slim_recipe_specs = {
+    "nonWF_recipe":              {"nonWF": True, "pedigree": True},
+    "long_nonWF_recipe":         {"nonWF": True},
+    "WF_recipe":                 {"WF": True, "pedigree": True},
+    "long_WF_recipe":            {"WF": True},
+    "WF_migration_recipe":       {"WF": True, "pedigree": True, "multipop": True},
+    "nonWF_early_recipe":        {"nonWF": True, "pedigree": True, "remembered_early": True},
+    "WF_early_recipe":           {"WF": True, "pedigree": True, "remembered_early": True},
+    "nucleotides_recipe":        {"WF": True, "pedigree": True, "nucleotides": True},
+    "long_nucleotides_recipe":   {"WF": True, "nucleotides": True},
+    "roots_recipe":              {"WF": True, "pedigree": True},
+    "nonWF_selfing_recipe":      {"nonWF": True, "pedigree": True},
+    "init_mutated_WF_recipe":    {"WF": True, "init_mutated": True},
+    "init_mutated_nonWF_recipe": {"nonWF": True, "init_mutated": True},
+    "with_metadata_recipe":      {"user_metadata": True},
+}
+
+# Some more complicated ones
+for t in ("WF", "nonWF"):
+    for s in ("early", "late"):
+        value = {t: True, "everyone": True, "pedigree": True}
+        if s == 'early':
+            value['remembered_early'] = True
+        basic_slim_recipe_specs[f'record_everyone_{t}_{s}_recipe'] = value
+
+
+# These SLiM scripts read in an existing trees file; the "input" gives a .trees files produced
+# by recipes above that is appropriate for starting this script.
+restarted_slim_recipe_specs = {
+    "nucleotides_restart":   {"WF": True, "nucleotides": True, "no_op": True, "input": f"tests/examples/recipe_nucleotides.trees"},
```

Comment:
this `input` looks different than the others

Reply:
Could be. I haven't tested this yet. I just want to get some feedback as to whether the general approach was OK.